### DISCUSSION TITLE
Efficient AD train_batch_size problem

### DISCUSSION THREAD

**Discussion by Balporsugu21** [open-edge-platform/anomalib#2232] [Q&A]
from anomalib.data import MVTec
from anomalib.models import EfficientAd
from anomalib.engine import Engine
from anomalib.models import Ganomaly
from anomalib.utils.normalization import NormalizationMethod
from anomalib.data import Folder
from anomalib.data.image.folder import FolderDataset
from anomalib.metrics import F1Score
from anomalib import TaskType

datamodule = MVTec()
model = EfficientAd()

engine = Engine(
    normalization=NormalizationMethod.MIN_MAX,
    threshold="F1AdaptiveThreshold",
    task=TaskType.CLASSIFICATION,

    accelerator="auto",
    check_val_every_n_epoch=1,
    limit_train_batches=1.0,
    limit_val_batches=1.0,
    limit_test_batches=1.0,
    precision=32,
    log_every_n_steps=20,
    devices=1,
    max_epochs=10,
    num_sanity_val_steps=0,
    val_check_interval=1.0,
)
engine.fit(datamodule=datamodule, model=model)

test_results = engine.test(
    model=model,
    datamodule=datamodule,
    ckpt_path=engine.trainer.checkpoint_callback.best_model_path,
)
After writing this code, I get this error. What should I do?

 ValueError: train_batch_size for EfficientAd should be 1.

**Comment by alexriedel1** [comment]
`datamodule = MVTec(train_batch_size=1)`

**Reply by Balporsugu21** [reply]
Thank you, but now I got this error, what should I do? RecursionError: maximum recursion depth exceeded while calling a Python object

**Reply by alexriedel1** [reply]
Can you show the full error log and your full code?

**Reply by Balporsugu21** [reply]
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/tqdm/std.py", line 1148, in __del__
    self.close()
  File "/usr/local/lib/python3.10/dist-packages/tqdm/std.py", line 1277, in close
    if self.last_print_t < self.start_t + self.delay:
AttributeError: 'DownloadProgressBar' object has no attribute 'last_print_t'
---------------------------------------------------------------------------
RecursionError                            Traceback (most recent call last)
[/usr/local/lib/python3.10/dist-packages/rich/console.py](https://localhost:8080/#) in print(self, sep, end, style, justify, overflow, no_wrap, emoji, markup, highlight, width, height, crop, soft_wrap, new_line_start, *objects)
   1673         with self:
-> 1674             renderables = self._collect_renderables(
   1675                 objects,

37 frames
RecursionError: maximum recursion depth exceeded while calling a Python object

During handling of the above exception, another exception occurred:

RecursionError                            Traceback (most recent call last)
... last 10 frames repeated, from the frame below ...

[/usr/local/lib/python3.10/dist-packages/rich/file_proxy.py](https://localhost:8080/#) in flush(self)
     51         output = "".join(self.__buffer)
     52         if output:
---> 53             self.__console.print(output)
     54         del self.__buffer[:]
     55 

RecursionError: maximum recursion depth exceeded while calling a Python objcet. my full code. Link to my entire code: https://colab.research.google.com/drive/1VXajUhaZn88bur0ppDmLDuqxga8MT-8K#scrollTo=3Fvrqd2CbWpZ

**Reply by Balporsugu21** [reply]
I got the same error on the Cfa model

**Reply by alexriedel1** [reply]
Yeah you are probably using anomalib in a jupyter notebook? The issue is known https://github.com/openvinotoolkit/anomalib/issues/2145 and for now you would have to build yourself a workaround

**Reply by Balporsugu21** [reply]
> Yeah you are probably using anomalib in a jupyter notebook? The issue is known #2145 and for now you would have to build yourself a workaround

I use Anomalybi from Google Colab. I was able to run it 3 months ago, but unfortunately it doesn't work now. What can I do? This is the colab link to run it. https://colab.research.google.com/drive/1VXajUhaZn88bur0ppDmLDuqxga8MT-8K#scrollTo=xTUkvyUbbapC
